src/main/scala/org/ua/ilm/datastore/io/reader/IdChannelsReader.scala

```diff
@@ -0,0 +1,15 @@
+package org.ua.ilm.datastore.io.reader
+
+import java.io.File
+
+
+class IdChannelsReader(file:File) extends FileReader(file){
```

Comment:
I think there is no need in separation. FileReader is enough. To list logic may be implemented via simple lambda whenever you want